src/Parser.ts

```diff
@@ -17,35 +17,44 @@ class ParserError extends Error {}
 export class Parser {
   public index: number = 0
 
-  constructor(
-    public tokens: Token[],
-  ) {}
+  constructor(public tokens: Token[]) {}
 
   parse(): Expression {
     const expr = this.expression()
+    if (this.isAtEnd() == false) {
+      throw new ParserError('unparsed tokens')
```

Comment:
can you make this error a little more descriptive?

Reply:
Got that updated with a more descriptive error.